File: 8kyu/020722.py
# https://www.codewars.com/kata/56676e8fabd2d1ff3000000c
# Level 8kyu

## Directions ##

# Can you find the needle in the haystack?
#
# Write a function findNeedle() that takes an array full of junk but containing one "needle"
# After your function finds the needle it should return a message (as a string) that says:
# "found the needle at position " plus the index it found the needle, so:
# find_needle(['hay', 'junk', 'hay', 'hay', 'moreJunk', 'needle', 'randomJunk'])
# should return "found the needle at position 5"

## Function ##

def find_needle(haystack):
    for i, needle in enumerate(haystack):
        if needle == 'needle':
            print('found the needle at position ' + str(i))
            return 'found the needle at position ' + str(i)



## Test Cases ##

find_needle(['3', '123124234', None, 'needle', 'world', 'hay', 2, '3', True, False])
# =>'found the needle at position 3'
find_needle(['283497238987234', 'a dog', 'a cat', 'some random junk', 'a piece of hay', 'needle', 'something somebody lost a while ago'])
# => 'found the needle at position 5'
find_needle([1,2,3,4,5,6,7,8,8,7,5,4,3,4,5,6,67,5,5,3,3,4,2,34,234,23,4,234,324,324,'needle',1,2,3,4,5,5,6,5,4,32,3,45,54])
# => 'found the needle at position 30'

File: 8kyu/020722.go
package main

import "fmt"

func main() {
	FindNeedle([]interface{}{"3", "123124234", nil, "needle", "world", "hay", 2, "3", true, false}, "needle")
	FindNeedle([]interface{}{"283497238987234", "a dog", "a cat", "some random junk", "a piece of hay", "needle", "something somebody lost a while ago"}, "needle")
	FindNeedle([]interface{}{1, 2, 3, 4, 5, 6, 7, 8, 8, 7, 5, 4, 3, 4, 5, 6, 67, 5, 5, 3, 3, 4, 2, 34, 234, 23, 4, 234, 324, 324, "needle", 1, 2, 3, 4, 5, 5, 6, 5, 4, 32, 3, 45, 54}, "needle")
}

func FindNeedle(haystack []interface{}, needle string) int {
	for i, v := range haystack {
		if v == needle {
			fmt.Println(i)
			return i
		}
	}
	return -1
}

// Test Cases

//package kata_test
//
//import (
//	. "github.com/onsi/ginkgo"
//	. "github.com/onsi/gomega"
//	//. "codewarrior/kata" (needed in website to run tests)
//)
//var _ = Describe("Test Example", func() {
//	It("should test example values", func() {

//	})
//})
//
//// Sample Test Cases
//// package kata_test (needed in website to run sample tests but already imported above)
////
////import (
//	//. "github.com/onsi/ginkgo"
//	//. "github.com/onsi/gomega"
//	//. "codewarrior/kata"
////)
//
//var _ = Describe("Test Example", func() {
//	It("should test example values", func() {

//	})
//})
